### ISSUE TITLE
[jvm package] xgboost model training early stop at minimum auc value

### ISSUE THREAD

**Issue by henghamao** [dmlc/xgboost#9377]
jvm package cited in pom.xml:
```
<dependency>
            <groupId>ml.dmlc</groupId>
            <artifactId>xgboost4j_2.12</artifactId>
            <version>1.7.6</version>
  </dependency> 
```
The model training code:
```
public void train(DMatrix trainMatrix){
        int round = 100; 
        int earlyStop = 10;
            
        HashMap<String, Object> mapParams = new HashMap<>(); 
        mapParams.put("eta", 0.1); 
        mapParams.put("objective", "binary:logistic");
        mapParams.put("max_depth", 3); 
        mapParams.put("eval_metric", "auc"); 
        mapParams.put("silent", 0);
        
        HashMap<String, DMatrix> mapWatches = new HashMap<>(); 
        mapWatches.put("selTrain", trainMatrix ); 

        try{
            booster = XGBoost.train(trainMatrix, mapParams, round, mapWatches, null, null, null, earlyStop); 
            System.out.println(booster.getAttrs() ); //打印训练的结果
            saveModel();
        }
        catch (Exception e) {
            e.printStackTrace();
            log.error("Error to train XGBoost Model", e.getMessage());
        }
    }
```
The model stop at the fist round which had the minimum auc value, while we expected the model to stop at the maximun auc value.

**Comment by trivialfis** [comment]
cc @wbo4958

**Comment by wbo4958** [comment]
sure, I will check it.

**Comment by wbo4958** [comment]
@henghamao, could you add `mapParams.put("maximize_evaluation_metrics", true);` and retry?

**Comment by henghamao** [comment]
Thanks！
By adding the line "mapParams.put("maximize_evaluation_metrics", true);", the training stopped at the maxmize AUC.
```
[10:07:09] [91] selTrain-auc:0.61312234112458341

[10:07:09] [92] selTrain-auc:0.61317036169818728

[10:07:09] [93] selTrain-auc:0.61344198401778538

[10:07:09] [94] selTrain-auc:0.61385812077231050

[10:07:09] [95] selTrain-auc:0.61395870490803284

[10:07:09] [96] selTrain-auc:0.61397982565356402

[10:07:09] [97] selTrain-auc:0.61431707070547215

[10:07:10] [98] selTrain-auc:0.61436486693250414

[10:07:10] [99] selTrain-auc:0.61446896184963551

{best_score=0.61446893, best_iteration=99}
Test data length:305334
```
And I thought it might be more convenient for the code to autmatically decide whether to maximize or minimize the metric. Python package is able to automatically maximize auc metric for the training.

**Comment by wbo4958** [comment]
@henghamao Good suggestion, Let me try to figure out how to support it.
